src/extensions/default/NavigationAndHistory/main.js

```diff
@@ -0,0 +1,464 @@
+        DocumentManager     = brackets.getModule("document/DocumentManager"),
+        EditorManager       = brackets.getModule("editor/EditorManager"),
+        ProjectManager      = brackets.getModule("project/ProjectManager"),
+        CommandManager      = brackets.getModule("command/CommandManager"),
+        Commands            = brackets.getModule("command/Commands"),
+        Dialogs             = brackets.getModule("widgets/Dialogs"),
+        Menus               = brackets.getModule("command/Menus"),
+        FileSystem          = brackets.getModule("filesystem/FileSystem"),
+        FileUtils           = brackets.getModule("file/FileUtils"),
+        ViewUtils           = brackets.getModule("utils/ViewUtils"),
+        WorkingSetView      = brackets.getModule("project/WorkingSetView"),
+        PreferencesManager  = brackets.getModule("preferences/PreferencesManager"),
+        KeyBindingManager   = brackets.getModule("command/KeyBindingManager"),
+        ExtensionUtils      = brackets.getModule("utils/ExtensionUtils");
+    
+    // Command constants for recent files
+    var SHOW_RECENT_FILES       = "show.recent.files",
+        NEXT_IN_RECENT_FILES    = "next.recent.files",
+        PREV_IN_RECENT_FILES    = "prev.recent.files";
+    
+    var htmlTemplate = require("text!html/recentfiles-template.html"),
+        dirtyDotTemplate = "<div class='file-status-icon dirty' style='position: absolute;margin-left: -2px;'></div>";
+    
+    // Delay in ms for hide timer when open recent files dialog shown from keyborad only commands
+    var HIDE_TIMEOUT_DELAY = 1500,
+        OPEN_FILE_DELAY    = 600;
+    
+    /*
+    * Contains list of most recently opened files and their last known cursor position
+    * @private
+    * @type {Array.<Object>}
+    */
+    var _mrofList = [],
+        $mrofContainer = null,
+        _activePaneId = null;
+    
+    
+    var $currentContext,
+        hideTimeoutVar,
+        openFileTimeoutVar,
+        activeEditor;
+    
+    var _hideMROFList;
+    
+    /**
+     * Opens a full editor for the given context
+     * @private
+     * @param {Object.<path, paneId, cursor>} contextData - wrapper to provide the information required to open a full editor
+     * @return {$.Promise} - from the commandmanager 
+     */
+    function _openEditorForContext(contextData) {
+        // Open the file in the current active pane to prevent unwanted scenarios, fallback to the persisted paneId
+        // only when unable to determine active paneId (like empty working set at the launch)
+        var activePaneId = MainViewManager.getActivePaneId();
+        return CommandManager.execute(Commands.FILE_OPEN, {fullPath: contextData.path,
+                                                    paneId: activePaneId || contextData.paneId }).done(function () {
+            activeEditor = EditorManager.getActiveEditor();
+            activeEditor.setCursorPos(contextData.cursor);
```

Comment:
Yes @nethip. Assume that this file was not in mru list. Then Brackets will put cursor on (1,1) , but as the context data was persisted and when the file was closed, we try to put back the cursor to the same place where it was exactly before the file was closed.